kasa/iot/modules/dimmer.py

```diff
@@ -0,0 +1,245 @@
+                icon="mdi:clock-out",
+                attribute_getter="gentle_on_time",
+                attribute_setter="set_gentle_on_time",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_ramp_rate",
+                name="Dimmer Ramp Rate",
+                icon="mdi:clock-fast",
+                attribute_getter="ramp_rate",
+                attribute_setter="set_ramp_rate",
+                type=Feature.Type.Switch,
+                category=Feature.Category.Config,
+            )
+        )
+
+    def query(self) -> dict:
+        """Request Dimming configuration."""
+        req = merge(
+            self.query_for_command("get_dimmer_parameters"),
+            self.query_for_command("get_default_behavior"),
+        )
+
+        return req
+
+    @property
+    def config(self) -> dict:
+        """Return current configuration."""
+        return self.data["get_dimmer_parameters"]
+
+    @property
+    def threshold_min(self) -> int | None:
+        """Return the minimum dimming level for this dimmer."""
+        if (min := self.config.get("minThreshold")) is not None:
+            return int(min)
+        return None
+
+    async def set_threshold_min(self, min: int) -> dict:
+        """
+        Set the minimum dimming level for this dimmer.
+
+        The value will depend on the luminaries connected to the dimmer.
+
+        :param min: The minimum dimming level, in the range 0-51.
+        """
+        if (min < 0) or (min > 51):
+            raise KasaException(
+                "Minimum dimming threshold is outside the supported range: 0-51."
+            )
+        return await self.call("calibrate_brightness", {"minThreshold": min})
+
+    @property
+    def fade_off_time(self) -> int | None:
+        """Return the fade off animation duration."""
+        if (fade_time := self.config.get("fadeOffTime")) is not None:
+            return int(fade_time)
+        return None
+
+    async def set_fade_off_time(self, time: int) -> dict:
+        """
+        Set the duration of the fade off animation.
+
+        :param time: The animation duration, in ms.
+        """
+        if (time < 0) or (time > 10_000):
+            # FYI:  Not sure if there is really a max bound here,
+            #       but anything above 10s seems ridiculous.
+            raise KasaException(
+                "Fade time is outside the bounds of the supported range: 0-10,000."
+            )
+        return await self.call("set_fade_on_time", {"fadeTime": time})
+
+    @property
+    def fade_on_time(self) -> int | None:
+        """Return the fade on animation duration."""
+        if (fade_time := self.config.get("fadeOnTime")) is not None:
+            return int(fade_time)
+        return None
```

Comment:
```suggestion
    def fade_on_time(self) -> timedelta:
        """Return the fade on animation duration."""
        return timedelta(milliseconds=self.config["fadeOnTime"])
```
It's a good idea to define the durations using `timedelta`, as this will allow downstreams (and our cli) to format the output without the need to define the unit.

If we can also expect that the values are always available, we can drop `| None` to avoid the need for downstreams to check for the type.

Reply:
I don't know enough about the API to say for certain that the values with absolutely always be there, but based on the existing fixtures, it seems like that's pretty likely to be true. I'll drop the `None` from the return type and convert to timedeltas.

Reply:
Do the setters need to accept `timedelta` on top of `int` or is just `int` ok for the setters?

Reply:
So far we haven't had timedeltas as an input as that's probably much more complicated than allowing just integers.

I'd say that we keep using integers for now with using the same precision as used by the device. We can always revisit that and add support for timedeltas afterwards, if necessary.

Reply:
On this one I’d say if we return a time delta we should allow a time delta for setters too. Perhaps it’s best to accept both int and timedelta and covert as necessary.

Reply:
In order to make the comparison against the timedelta bound constants, I wound up changing the setters to accept `int | timedelta` and then just wrap `int` as a `timedelta(milliseconds=int)` to clean it up.